dvc/tree/oss.py

```diff
@@ -116,6 +116,21 @@ def remove(self, path_info):
         logger.debug(f"Removing oss://{path_info}")
         self.oss_service.delete_object(path_info.path)
 
+    def upload_fobj(self, fobj, to_info, no_progress_bar=False, **pbar_args):
+        with Tqdm.wrapattr(
+            fobj, "read", disable=no_progress_bar, bytes=True, **pbar_args
+        ) as wrapped:
```

Comment:
Sucks how we have to repeat the pbar logic in every implementation. It is clear that in the end we will do what fsspec does and create an AbstractFile, so that pbar could be used once up the stack. Not quite sure it is worth to bother with right now...

Reply:
Should I hoist the upload_fobj into the BaseTree to manage the progress bar, or just leave it (just like upload/download)?

Reply:
@isidentical If you can extract it like that with a reasonable amount of effort it would indeed be amazing! If not - we can leave it as is.